src/arch/arm/64/kernel/vspace.c

```diff
@@ -2406,7 +2405,8 @@ exception_t decodeARMMMUInvocation(word_t invLabel, word_t length, cptr_t cptr,
 
         /* Find first free ASID */
         asid = cap_asid_pool_cap_get_capASIDBase(cap);
-        for (i = 0; i < (1 << asidLowBits) && (asid + i == 0 || pool->array[i]); i++);
+        for (i = 0; i < (1 << asidLowBits) && (asid + i == 0
+                                               || (asid_map_get_type(pool->array[i]) != asid_map_asid_map_none)); i++);
```

Comment:
If for some reason the previous shorter version is preferred, I think it would be Ok to leave it, because it should be equivalent. My guess is that `-O2` will reduce it to the same assembly anyway, though, and this is slightly clearer.

Reply:
Yes, you're right, that doesn't really work. I tend to ignore C's type system because it doesn't guarantee anything :-) I was only looking at the representation, which hasn't changed for the False case.